plugins/gin/gin.go

```diff
@@ -33,11 +34,47 @@ const (
 	httpServerComponentID int32 = 49
 )
 
+type routeInfo struct {
+	operationName string
+}
+
+type middleware struct {
+	routeMap     map[string]map[string]routeInfo
+	routeMapOnce sync.Once
+}
+
 //Middleware gin middleware return HandlerFunc  with tracing.
-func Middleware(tracer *go2sky.Tracer) gin.HandlerFunc {
-	return func(c *gin.Context) {
+func Middleware(engine *gin.Engine, tracer *go2sky.Tracer) gin.HandlerFunc {
+	if engine == nil || tracer == nil {
+		return nil
+	}
+	m := new(middleware)
 
-		operationName := fmt.Sprintf("/%s%s", c.Request.Method, c.HandlerName())
+	return func(c *gin.Context) {
+		m.routeMapOnce.Do(func() {
+			routes := engine.Routes()
+			rm := make(map[string]map[string]routeInfo)
+			for _, r := range routes {
+				mm := rm[r.Method]
+				if mm == nil {
+					mm = make(map[string]routeInfo)
+					rm[r.Method] = mm
+				}
+				mm[r.Handler] = routeInfo{
+					operationName: fmt.Sprintf("/%s%s", r.Method, r.Path),
+				}
+				m.routeMap = rm
+			}
+		})
+		var operationName string
+		handlerName := c.HandlerName()
+		if routeInfo, ok := m.routeMap[c.Request.Method][handlerName]; ok {
+			operationName = routeInfo.operationName
+		}
+		if operationName == "" {
```

Comment:
You probably want still to trace this under the `not_found` span name.

Reply:
If the operation name is empty, do you still need to track?
@wu-sheng what are you thinking？

Reply:
in such cases operation names usually are `GET not_found` or just `GET` (change the verb accordingly) but @wu-sheng might provide more idiomatic feedback for skywalk

Reply:
According to the current GO Agent API definition,  `operationName` should not be an empty string(illegal right now). Right? @hanahmily 

I think `GET` should be a better name.

Reply:
```GET```? maybe  ```GET/unknown```  is more appropriate.

Reply:
unknown is a uri, which may cause misguiding if user really have /unknown.

Keep it empty is safer from my perspective.

Reply:
Thank you for your reply.
too busy. it already fix.